// coffeelint/lib/rules/max_line_length.js
// Generated by CoffeeScript 1.6.2
(function() {
  var MaxLineLength, regexes;

  regexes = {
    longUrlComment: /^\s*\#\s*http[^\s]+$/
  };

  module.exports = MaxLineLength = (function() {
    function MaxLineLength() {}

    MaxLineLength.prototype.rule = {
      name: 'max_line_length',
      value: 80,
      level: 'error',
      message: 'Line exceeds maximum allowed length',
      description: "This rule imposes a maximum line length on your code. <a\nhref=\"http://www.python.org/dev/peps/pep-0008/\">Python's style\nguide</a> does a good job explaining why you might want to limit the\nlength of your lines, though this is a matter of taste.\n\nLines can be no longer than eighty characters by default."
    };

    MaxLineLength.prototype.lintLine = function(line, lineApi) {
      var max, _ref;

      max = (_ref = lineApi.config[this.rule.name]) != null ? _ref.value : void 0;
      if (max && max < line.length && !regexes.longUrlComment.test(line)) {
        return {
          context: "Length is " + line.length + ", max is " + max
        };
      }
    };

    return MaxLineLength;

  })();

}).call(this);
